neuralop/layers/spectral_convolution.py

```diff
@@ -390,6 +383,21 @@ def transform(self, x, layer_index=0, output_shape=None):
                 list(range(2, x.ndim)),
                 output_shape=out_shape,
             )
+    
+    @property
+    def n_modes(self):
+        return self._n_modes
+    
+    @n_modes.setter
+    def n_modes(self, n_modes):
+        if isinstance(n_modes, int): # Should happen for 1D FNO only
+            n_modes = [n_modes]
+        else:
+            n_modes = list(n_modes)
```

Comment:
Should we have more explicit type checking here? Something like: 
```suggestion
        if isinstance(n_modes, int): # Should happen for 1D FNO only
            n_modes = [n_modes]
        elif all([isinstance(m, int) for m in n_modes]):  # Assumes n_modes is a populated iterable
            n_modes = list(n_modes)
        else:
            raise ValueError(msg)
```
Otherwise a model choking on `n_modes` further in execution could be pretty confusing.

Reply:
I assumed it was clear modes should be int, it's also in the docstring: I don't check if it's an int for type check - just to allow users to call FNO(mode, ...) for the 1D FNO rather than FNO( (mode,), ...)